xarray/core/weighted.py

```diff
@@ -239,6 +292,123 @@ def _weighted_std(
 
         return cast("DataArray", np.sqrt(self._weighted_var(da, dim, skipna)))
 
+    def _weighted_quantile(
+        self,
+        da: "DataArray",
+        q: Union[float, Iterable[float]],
+        dim: Union[Hashable, Iterable[Hashable]] = None,
+        skipna: Optional[bool] = None,
+        keep_attrs: Optional[bool] = None,
+    ) -> "DataArray":
+        """Apply a weighted ``quantile`` to a DataArray along some dimension(s)."""
+
+        def _get_h(
+            n: float, q: np.ndarray, htype: Literal[7, 4, 5, 6, 8, 9] = 7
```

Comment:
I know there are several names for "the 9" and these numbers refer to R implementation. But, np tried to gives them proper name, it could be good to keep these names here ?

Dr. Hyndman proposed theses names https://github.com/numpy/numpy/issues/10736#issuecomment-898151920 
np now follows these, with the exception of the default, still being named "linear".

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.quantile.html

Reply:
Added support for both explicit names and numbers.

Reply:
Any particular reason for leaving the integers as well? These are new (non-public?) methods so backwards compat shouldn't be a concern. If it's just for referencing the R variants I think it can be mentioned in the docstring or comments instead.

Reply:
Just convenience, I thought some people might be more familiar with the numbering system than the numpy convention. Granted none of this is exposed at the moment, but I thought I'd make it as effortless as possible to expose them later on. I'm not strongly attached to this, if you say you don't want the numbers, I'll remove them.

Reply:
I prefer simply mirroring numpy, then no decisions have to be made! 
I think it will be enough to explain which number the string corresponds to in the docstring like numpy does.

Reply:
Done.